server/src/hardware/protocol/z21/clientkernel.cpp

```diff
@@ -112,6 +113,44 @@ void ClientKernel::receive(const Message& message)
               });
           }
           break;
+
+        case LAN_X_LOCO_INFO:
+        {
+          const int n = message.dataLen() - 7;
+          if(n >= 7 && n <= 14)
+          {
+            const auto& reply = static_cast<const LanXLocoInfo&>(message);
+
+            bool val[31 + 1] = {};
+            const int maxFunc = (n >= 8) ? 31 : 28;
```

Comment:
suggestion: add a method to `LanXLocoInfo` to get the max function number, if the message is extended again in the future, this will still work :)
And a static constexpr with the message maximum for the `bool val[]` array above :)